### DISCUSSION TITLE
Help me understand versions in Pimcore docs

### DISCUSSION THREAD

**Discussion by NiklasBr** [pimcore/pimcore#15832] [General]
![image](https://github.com/pimcore/pimcore/assets/279826/1db0c357-6c73-40be-b8df-716f09df1a78)

Opening the URL https://pimcore.com/docs redirects to https://pimcore.com/docs/platform/Pimcore/ which is tagged as *Version: 2023.1* but that is never referenced in [Packagist/Composer](https://packagist.org/packages/pimcore/pimcore) and the version selector on the [Docs site does not help either](https://pimcore.com/docs/platform/next/Platform_Version/Release_Notes/). Neither are releases here on `pimcore/pimcore` tagged with any reference to the `2023.0` or `2023.1` versions.

![image](https://github.com/pimcore/pimcore/assets/279826/7e10cef4-9a72-4f73-bef5-2451289ba759)   ![image](https://github.com/pimcore/pimcore/assets/279826/a400f5f1-fe99-40f2-afbe-d6374bc2f755)

Up until `10.6` it was very easy to find the documentation for each version, this was crucial when for example updating from `10.2` to `10.5` or when providing support for an installation still on `10.4` for example.

Since Pimcore claims to [adhere to semantic versioning](https://pimcore.com/en/resources/blog/latest-technology-and-best-performance-pimcore-x-is-out-now_a150669) and by the looks of it this is only a major.minor (without *dot patch* part of the scheme) it muddies the water somewhat regarding adherence since all [semantic versionning](https://semver.org/) MUST contain all three MAJOR.MINOR.PATCH numbers to be considered valid.

How are we supposed to find the correct documentation? Even as someone who develops for a Pimcore Enterprise customer this makes little sense since there is no way even for me to see the `YEAR.version` or list of "hidden" versions anywhere.

Where can Enterprise developers (and Community Edition developers for that matter) find the combined release notes of all bundles for version `2023.0` (it seems to be missing, see above)? And where for `2023.1` (which only appears to mention Pimcore 11 in general and not any exact version, or version range)?

**Comment by fashxp** [comment]
Hi folks, 
thank you very much for the post. We always appreciate feedback and improvement suggestions. Let me give some insights on that topic. 

One requirement for our new rendered documentation was to offer a global search including all content of all available modules. It turned out, with having a separate documentation versioning of all modules for their own, this is not possible. And that is one of the reasons, why we decided to version our rendered documentation with the newly introduced platform version. 

The platform version is an enterprise service which defines a set of specific versions of all Pimcore modules which work together. These versions are enforced via composer rules, further details are explained [here](https://pimcore.com/docs/platform/Platform_Version/). 
We started the platform version with the release of Pimcore Core Framework version 11 (2023.1), will release a new platform version 2023.2 next week and also will talk about it in our [upcoming webinar](https://app.livestorm.co/pimcore/stargazing-the-future-of-pimcore) next week. There will also be a platform version release for the LTS offering for Pimcore 10.


So, the platform version defines the minor version of all the modules of Pimcore we provide and subsequently also defines the available feature set. That is, why from our perspective it makes sense to version the documentation based on it too. Combined release notes are listed [here](https://pimcore.com/docs/platform/Platform_Version/Release_Notes/) and will contain detailed information about tasks needed for updating from one version to another - currently only one platform version exists. 

For simplicity and flexibility reasons, we decided to stick to two numbers (e.g. 2023.2) for platform version (knowing this is not strictly semantic versioning), and allow patch version updates of the modules within one minor platform version. We described that rule set in the [docs](https://pimcore.com/docs/platform/Platform_Version/#key-rules). 
The modules themselves (e.g. pimcore/pimcore, pimcore/data-hub, etc.) will still follow strict semantic versioning definition with three numbers. 
If that approach turns out to be not practical, we are ready to tag patch versions of the platform version too - that probably would slow down our release process of patch versions though. 

For enterprise users, we recommend to use platform version releases. Thus, it is easy to pick the corresponding version of the rendered documentation and you don't need to care about the specific versions of each module anymore. 

For community users, you can use the rendered documentation for general information. If you want to read specific docs version for a specific module, you always can switch to the corresponding github repository and read the docs there if needed (github does a very good job in nicely rendering markdown files). 
Also, the [old docs](https://pimcore.com/docs/pimcore/10.6/) are still reachable. If someone from the community is ready to maintain it (that also means to update daux and adapt to all the changes, include new releases, maintain on a regular base, etc.), we would be happy to merge PRs and include more current versions of the modules into the old docs. 


Have a good day...
